CONTRIBUTING.md

```diff
@@ -0,0 +1,417 @@
+
+- No direct committing to `main`, or `release/*`
+- Check json for validity
+- Ensure newline character at the end of files
+- Trim end of line whitespace characters
+- Check for no merge conflict lines accidentally being staged
+- Clang format
+- Python block (line length 100)
+
+
+## Testing and Analysis
+
+Several targets and CMake configuration options are defined as shown below to test the qe-compiler. The standard test suite consists of a format check to ensure the source follows the configured style defined in `.clang-format`, a run of `clang-tidy` to perform code analysis, and a run of the test suite. It is also possible to run additional analysis with Valgrind and the Google Sanitizers below. CMake commands should always be run from a build directory.
+
+Our test suite contains both unit tests (using googletest) and integration tests that exercise the full qss-compiler or qss-opt tools (using LLVM lit).
+
+### Debugging LIT Tests
+
+A large fraction of the regression test suite is orchestrated using the LLVM Integrated Tester (LIT). When investigating test execution, several command line arguments for LIT may be helpful:
+
+- `-vv` or `--echo-all-commands` to print all commands to stdout while execution, which implies `-v` / `--verbose` that prints test output on test failures
+- `-a` or `--show-all` prints information about all tests during execution
+- `--show-suites` and `--show-tests` to list all discovered test suites and tests; note that these two options exit immediately and will not run the tests.
+
+There are several options for passing additional commands to lit:
+
+- set the cmake variable `LIT_TEST_EXTRA_ARGS` or include them directly in `test/CMakeLists.txt` in the call to `add_lit_testsuite` so that they will be used for the targets `check-tests` and `check-all`
+- run `ninja -v` and capture the command line  for `lit`, then modify and manually run that command line.
+
+LIT tests may be filtered in order to limit the number of tests that are run. LIT supports the
+`LIT_FILTER` and `LIT_FILTER_OUT` environment variables to filter in or out tests using a regex.
+For example using `export LIT_FILTER=Conversion/CodeGen/pulse-gen-drive-0.mlir` will limit the lit
+test to that single test. Note to use this with the `check-tests` target requires passing
+`-DLIT_TEST_EXTRA_ARGS=--allow-empty-runs` to cmake otherwise the testing will stop on the first
+target that has an empty run.
+
+### Setting Paths for Manual Test Runs
+
+The compiler may require static resources at runtime (e.g., target-specific tools or libraries). It will search for the resources in the path defined by the environment variable `QSSC_RESOURCES` or in the designated install location if that variable is not present.
+
+For running `qss-compiler` or `qss-opt` from the build directory, you may use a generated shell script that takes care of setting up the required environment variables as follows:
```

Comment:
I suppose this instance of `qss-compiler` has to stay?

Reply:
Yah, I tried to keep any tool mentions consistent for now.